### PR TITLE
rust 1.42.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by Bo98** [comment]
Cargo needs updating to 0.43.0 and the `cargobootstrap` resource needs updating to 0.42.0.

**Comment by Bo98** [comment]
The build will take several hours and there will be failures due to #50815. I'll cancel the current one until Cargo is updated.

**Comment by chenrui333** [comment]
Thanks! Yeah, I just updated the cargo version.

I was mostly thinking to create the PR with `brew bump-formula-pr` and updated the cargo from there.

File: Formula/rust.rb [hunk 1]
```diff
@@ -40,8 +40,8 @@ class Rust < Formula
 
   resource "cargobootstrap" do
     # From https://github.com/rust-lang/rust/blob/#{version}/src/stage0.txt
-    url "https://static.rust-lang.org/dist/2019-12-19/cargo-0.41.0-x86_64-apple-darwin.tar.gz"
-    sha256 "1ef77a6be5e697bb7dde40854651fc67e91f119d5a9ddf747a25e30c1179fbe1"
+    url "https://static.rust-lang.org/dist/2020-03-12/cargo-0.43.0-x86_64-apple-darwin.tar.gz"
+    sha256 "92d4c9fb4747dce158cdfb773651aea8eac894277f3a2de5aa2c3b9d92439d8e"
```
Comment (Bo98) [inline Formula/rust.rb]: No, this one should be 0.42.0 as per the comment.

https://github.com/rust-lang/rust/blob/1.42.0/src/stage0.txt

Comment (chenrui333) [reply Formula/rust.rb]: Somehow I could not find the link for `cargo-0.42.0-x86_64-apple-darwin.tar.gz`, feel free to update it.

Comment (Bo98) [reply Formula/rust.rb]: https://static.rust-lang.org/dist/2020-02-27/cargo-0.42.0-x86_64-apple-darwin.tar.gz

Comment (chenrui333) [reply Formula/rust.rb]: You are awesome!

Comment (chenrui333) [reply Formula/rust.rb]: Just curious, how did you find it?

I tried the date listed on the cargo github releases page, it did not quite work.

Comment (Bo98) [reply Formula/rust.rb]: I used the date in the stage0.txt file.

Comment (chenrui333) [reply Formula/rust.rb]: I see

**Comment by chenrui333** [comment]
# build failure
- `diesel`
- `gitg`
- `grin`
- `grin-wallet`
- `pijul`
- `rav1e`

# test failure
- `broot`

# shasum mismatch
- `gleam`

**Comment by chenrui333** [comment]
Give this another run.

**Comment by chenrui333** [comment]
gleam and diesel should be good now.

**Comment by chenrui333** [comment]
## catalina

diesel
efl
gitg
glade
gleam
grin
grin-wallet

## high-sierra

- diesel
- gitg
- glade
- grin
- grin-wallet

## mojave

diesel
gitg
glade
gleam
grin
grin-wallet

**Comment by chenrui333** [comment]
Filed Glade bug report in here, https://gitlab.gnome.org/GNOME/glade/-/issues/420

**Comment by Bo98** [comment]
Glade is our problem. docbook-xsl is currently broken after 1.79.2 was merged.

**Comment by chenrui333** [comment]
noted, do we have a fix PR for it?

**Comment by Bo98** [comment]
#52041

**Pull request closed without merging**